### PR TITLE
Redirect to HTTPS if enabled in the root page

### PR DESCRIPTION
**Pull request by aschempp**
see https://github.com/contao/contao/issues/262

### CODE REVIEW

File: core-bundle/src/Routing/Matcher/UrlMatcher.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+namespace Contao\CoreBundle\Routing\Matcher;
+
+use Symfony\Bundle\FrameworkBundle\Routing\RedirectableUrlMatcher;
+use Symfony\Cmf\Component\Routing\NestedMatcher\FinalMatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouteCollection;
+
```
Comment (Toflar) [inline core-bundle/src/Routing/Matcher/UrlMatcher.php:20]: This needs description, I have no clue what it is supposed to do.

Comment (aschempp) [reply core-bundle/src/Routing/Matcher/UrlMatcher.php:20]: description in regards to?

Comment (Toflar) [reply core-bundle/src/Routing/Matcher/UrlMatcher.php:20]: What the whole class does.

**Comment by Toflar** [comment]
Language file should be updated :)

File: core-bundle/src/Routing/Matcher/UrlMatcher.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Routing/Matcher/UrlMatcher.php:20]: Well it's the UrlMatcher. It extends from the one from the Symfony Framework Bundle, which implements redirecting, and it implements the interface required by Cmf Routing. That's all on the first line of the class, I don't see much else to describe. I don't see a point in adding extensive descriptions to framework functional classes, someone who can't read the code should not care about it anyway.

Comment (Toflar) [reply core-bundle/src/Routing/Matcher/UrlMatcher.php:20]: >  I don't see a point in adding extensive descriptions to framework functional classes, someone who can't read the code should not care about it anyway.

This is pure bullshit. But whatever.

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Comment by leofeyer** [comment]
@Toflar This is the current field name and description:



What would you like me to change?

**Comment by Toflar** [comment]
via HTTPS only? :)

**Comment by leofeyer** [comment]
Adjusted in 1541badfcb911906eec28b72d51f4254ada56b0e.

File: core-bundle/src/Routing/Matcher/UrlMatcher.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Routing/Matcher/UrlMatcher.php:20]: https://twitter.com/CodeWisdom/status/1090308606409478144

**Pull request merged**